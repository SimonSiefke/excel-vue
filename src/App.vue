<template>
  <div id="app">
    <h1>hello</h1>
    <button v-if="!loggedIn" @click="login">login</button>
    <p v-if="loggedIn">login successful</p>
    <button v-if="loggedIn" @click="getTimeEntries">get Time</button>
    {{timeEntries}}
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import axios from 'axios' // for ajax requests
import { TimeEntryKey, TimeEntry } from '@/types/TimeEntry'
import { buildTemplate } from '@/utils'
import ExcelAdapter from '@/utils/ExcelAdapter'

const url = process.env.VUE_APP_API_URL

interface Data {
  loggedIn: boolean
  timeEntries: TimeEntry[] | null
}

export default Vue.extend({
  name: 'app',
  data(): Data {
    return {
      loggedIn: false,
      timeEntries: null,
    }
  },
  methods: {
    async login() {
      const response = await axios.post(`${url}/login`)
      this.loggedIn = true
    },
    async getTimeEntries() {
      const response = await axios.get(`${url}/time`)
      this.timeEntries = response.data.result
    },
  },
  watch: {
    timeEntries(newTimeEntries) {
      if (newTimeEntries) {
        ExcelAdapter.writeToExcel(buildTemplate(newTimeEntries))
      }
    },
  },
})
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
