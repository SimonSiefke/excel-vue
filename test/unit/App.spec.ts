import { shallow, mount } from '@vue/test-utils'
import App from '@/App.vue'

describe('App.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallow(App)
    expect(wrapper.text()).toMatch('hello')
  })

  it('matches the snapshot', () => {
    const wrapper = mount(App)
    expect(wrapper.element).toMatchSnapshot()
  })
})
