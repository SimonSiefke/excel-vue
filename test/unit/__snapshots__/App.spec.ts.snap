// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue matches the snapshot 1`] = `
<div
  id="app"
>
  <h1>
    hello
  </h1>
   
  <button>
    login
  </button>
   
  <!---->
   
  <!---->
  
  

</div>
`;
